docs/adr/0002-delivery-channel-creation-and-integration.md

```diff
@@ -0,0 +1,81 @@
+2. Store user preferences somewhere that is accessible to the management api, then when a new type of notifications comes through, check what type of communications types  the user has agreed to use and then send a message through the delivery channel .
+
+The latter option was chosen as it provides a user the ability to choose what notifications they want out of the box, if the former is chosen, all users/businesses integrating with this service will then need to write their own notifications user management service.
+
+User settings for which notifications will go to what communication channels for each user will be stored inside a User Notifications Settings table which will be accessed by the Management Api. Configuration in this table will include the Notification Type and the appropriate communication types associated to that notification type. This table will have an associated crud api that will allow a user to view/change the stored settings.
+
+### Delivery Channels and Notification Types
+
+A delivery channel will need two types of setup, the first being the integration of the service with Marain and the second being configuration added in by a user to how this channel will be used in the current implementation (which notification types are permitted to which channels, templating, api keys etc).
+
+The management API will host configuration for what notification types can be sent through which delivery channels in the following format:
+
+```json
+[
+  {
+    "id": "",
+    "displayName": "",
+    "communicationTypes": [""],
+    "isConfigured": false
+  }
+]
+```
+where 'communicationTypes' specifies what communication types are supported in this Delivery Channel. This allows for consumers of Marain.UserNotifications to specify if they want to only use one kind communication type from a certain Delivery Channel that supports multiple types.
+
+A delivery channel will have a generic api that can be consumed by the Management Api and be able to transform data provided in the Notification into a format that is specific to that communication type (this will be discussed in a sequential adr). The result of this call will be a callback uri/null depending on how the delivery channel has been implemented.
+
+### Delivery Channel configuration 
+
+A delivery channel will also need the configuration of the underlying platform to be stored somewhere (api keys, secrets, etc). This will be stored in a generic manner so if a supported delivery channel needs to be used, an api key and slight modification of the settings in the management api will only be required. 
```

Comment:
When I refer to 'platform' here, I'm referring to the underlying notification provider (I'm not exactly sure what to call it as I've referred to it as a 'Delivery Channel' everywhere else). 

Yes, I was very vague when describing how the secrets will be stored.  (because that's what they are at the end of the day). I would also prefer to store them in a keyvault. I will adjust the document.